src/main/java/io/github/syst3ms/skriptparser/expressions/ExecExprReplace.java

```diff
@@ -0,0 +1,142 @@
+ * <b>replace</b> them from in the list. Note that not all expressions support this behaviour.
+ * If no valid match was found, nothing will happen.
+ * <br>
+ * Note that indices in Skript start at one.
+ *
+ * @name Replace
+ * @type EFFECT/EXPRESSION
+ * @pattern replace [(all|every|[the] first|[the] last)] [regex [pattern[s]]] %strings% in %strings% with %string%
+ * @pattern replace [(all|every|[the] first|[the] last)] [regex [pattern[s]]] %strings% with %string% in %strings%
+ * @since ALPHA
+ * @author Mwexim
+ * @see ExprElement
+ */
+public class ExecExprReplace extends ExecutableExpression<String> {
+	static {
+		Parser.getMainRegistration().addSelfRegisteringElement(
+				ExecExprReplace.class,
+				String.class,
+				false,
+				"replace [(0:all|0:every|1:[the] first|2:[the] last)] [4:regex [pattern[s]]] %strings% in %strings% with %string%",
+				"replace [(0:all|0:every|1:[the] first|2:[the] last)] [4:regex [pattern[s]]] %strings% with %string% in %strings%"
+		);
+	}
+
+	// 0 = all, 1 = first, 2 = last
+	private Expression<String> toMatch;
+	private Expression<String> toReplace;
+	private Expression<String> replacement;
+
+	private int type;
+	private boolean regex;
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContext parseContext) {
+		type = parseContext.getParseMark() & 0b11;
+		regex = (parseContext.getParseMark() ^ type) == 4;
+
+		toMatch = (Expression<String>) expressions[0];
+		toReplace = (Expression<String>) expressions[1 + matchedPattern];
+		replacement = (Expression<String>) expressions[2 - matchedPattern];
+
+		if (!toReplace.acceptsChange(ChangeMode.SET, String[].class)) {
+			var logger = parseContext.getLogger();
+			logger.error(
+					"The expression '"
+							+ toReplace.toString(TriggerContext.DUMMY, logger.isDebug())
+							+ "' cannot be changed",
+					ErrorType.SEMANTIC_ERROR
+			);
+			return false;
+		}
+		return true;
+	}
+
+	@Override
+	public String[] getValues(TriggerContext ctx, boolean isEffect) {
+		String[] replacedValues = toReplace.getValues(ctx);
+		String[] matchedValues = toMatch.getValues(ctx);
+		String replacementValue = replacement.getSingle(ctx).orElse(null);
+		if (replacementValue == null) {
+			return replacedValues;
+		}
+
+		for (int i = 0; i < replacedValues.length; i++) {
+			for (String match : matchedValues) {
+				String current = replacedValues[i];
+				String replaced;
+				if (regex) {
+					// The regex pattern must be valid
+					try {
+						Pattern.compile(match);
+					} catch (PatternSyntaxException ignored) {
+						continue;
+					}
+				}
+				switch (type) {
+					case 0:
+						replaced = regex
+								? current.replaceAll(match, replacementValue)
+								: current.replace(match, replacementValue);
+						break;
+					case 1:
+						replaced = current.replaceFirst(
+								regex ? match : Pattern.quote(match),
+								replacementValue
+						);
+						break;
+					case 2:
```

Comment:
Nothing to change here, I just want to show a point here. I tried to understand why using the ``.*`` before the match, because I don't think it's needed. But here is the problem I got: I need to find what's 2 and I'm losing time to understand that. That's why I asked a long time ago to use enums instead, but it seems like it was not agreed.

Reply:
I will add comments to make the code clearer, as I can see how it can lead to confusion.